### PR TITLE
drivers/serial: stm32: Modify Kconfig instance flags

### PR DESCRIPTION
**Pull request by erwango**
STM32 uart Kconfig instance flags were not following
same naming scheme than other drivers (i2c, spi, ..)

Update driver to use UART_X instead of UART_STM32_PORT_X

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

### CODE REVIEW

**Comment by erwango** [comment]
Shippable timed out.
I tested beforehand with sanitycheck with all boards on hello_world. 
So I suspect timeout is due to impact of change..
Let my try once more

**Comment by erwango** [comment]
recheck

**Comment by erwango** [comment]
recheck

File: boards/arm/96b_neonkey/Kconfig.defconfig [hunk 1]
```diff
@@ -12,8 +12,8 @@ config BOARD
 
 if UART_CONSOLE
 
-config UART_STM32_PORT_1
-	default y
+config UART_1
+	def_bool y
```
Comment (nashif) [inline boards/arm/96b_neonkey/Kconfig.defconfig]: default y, see 46386522142e546c6e028db0f560c6db25a02d06

**Review by nashif — DISMISSED** [review]

File: boards/arm/96b_neonkey/Kconfig.defconfig (hunk 1 above)
Comment (erwango) [reply boards/arm/96b_neonkey/Kconfig.defconfig]: Txs, rebase issue I guess. Fixed

**Comment by erwango** [comment]
@nashif, comments addressed

**Pull request merged**